### PR TITLE
[CALCITE-5961] Type inference of ARRAY_COMPACT is incorrect

### PR DESCRIPTION
**Pull request by chucheng92**
https://issues.apache.org/jira/browse/CALCITE-5961

### CODE REVIEW

File: testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java [hunk 1]
```diff
@@ -5885,7 +5885,9 @@ private static void checkIf(SqlOperatorFixture f) {
 
     final SqlOperatorFixture f = f0.withLibrary(SqlLibrary.SPARK);
     f.checkScalar("array_compact(array[null, 1, null, 2])", "[1, 2]",
-        "INTEGER ARRAY NOT NULL");
+        "INTEGER NOT NULL ARRAY NOT NULL");
+    f.checkScalar("array_compact(array[1, 2])", "[1, 2]",
+        "INTEGER NOT NULL ARRAY NOT NULL");
     f.checkScalar("array_compact(array[null])", "[]",
         "NULL ARRAY NOT NULL");
     f.checkScalar("array_compact(array(null))", "[]",
```
Comment (NobiGo) [inline testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java:5893]: Two same tests for `array_compact(array[null])`?

Comment (chucheng92) [reply testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java:5893]: @NobiGo thank for reviewing. Yes, it was introduced by the previous commit/PR, and I can remove it in this PR? WDYT?

**Review by NobiGo — APPROVED** [review]

**Comment by chucheng92** [comment]
@NobiGo hi. xiong. thanks for reviewing. 
It seems that the commit history has increased a lot due to a force-push by Tanner. Do I need to rebase the main and then wait to see if there are other reviews, then we can merge this?

**Comment by chucheng92** [comment]
hi, @NobiGo. If there are no new reviewers to give a comment. Would you help to merge this? thanks.

**Pull request merged**